### PR TITLE
Backports 0.18 pr12

### PR DESCRIPTION
**Pull request by Munkybooty**

### CODE REVIEW

File: doc/build-osx.md [hunk 1]
```diff
@@ -51,7 +51,7 @@ Running
 
 Dash Core is now available at `./src/dashd`
 
-Before running, it's recommended that you create an RPC configuration file.
+Before running, you may create an empty configuration file.
 
     echo -e "rpcuser=dashrpc\nrpcpassword=$(xxd -l 16 -p /dev/urandom)" > "/Users/${USER}/Library/Application Support/DashCore/dash.conf"
```
Comment (PastaPastaPasta) [inline doc/build-osx.md:56]: 14684 this needs changing

https://github.com/bitcoin/bitcoin/pull/14684/files#diff-e9ba5bd8b63e72aa2d54d02960c9209823938981691cc8786ba323b54de0659cL85-R85

File: contrib/debian/examples/dash.conf [hunk 2]
```diff
@@ -122,21 +119,21 @@
 # running on another host using this option:
 #rpcconnect=127.0.0.1
 
+# Wallet options
+
 # Create transactions that have enough fees so they are likely to begin confirmation within n blocks (default: 6).
 # This setting is over-ridden by the -paytxfee option.
 #txconfirmtarget=n
 
+# Pay a transaction fee every time you send bitcoins.
```
Comment (PastaPastaPasta) [inline contrib/debian/examples/dash.conf:128]: 14684 dash

File: src/qt/optionsdialog.cpp [hunk 3]
```diff
@@ -302,6 +310,15 @@ void OptionsDialog::setMapper()
...
 
+    /* Window */
+#ifndef Q_OS_MAC
+    if (QSystemTrayIcon::isSystemTrayAvailable()) {
+        mapper->addMapping(ui->hideTrayIcon, OptionsModel::HideTrayIcon);
+        mapper->addMapping(ui->minimizeToTray, OptionsModel::MinimizeToTray);
+    }
+    mapper->addMapping(ui->minimizeOnClose, OptionsModel::MinimizeOnClose);
+#endif
+
```
Comment (PastaPastaPasta) [inline src/qt/optionsdialog.cpp:321]: This adds too much... Maybe missing a prior backport?

https://github.com/bitcoin/bitcoin/pull/14228/files#diff-1d6f5264407b3b2e18be8d9d9653bbf1d67979f4f95dc26575dc23bb8463b4e7L214-R225

File: doc/release-notes-14060.md [hunk 4]
```diff
@@ -0,0 +1,21 @@
+Configuration
```
Comment (PastaPastaPasta) [inline doc/release-notes-14060.md:1]: drop 14704 and mark DNM

File: .travis.yml [hunk 5]
```diff
@@ -1,8 +1,3 @@
-# errata:
-# - sudo/dist/group are set so as to get Blue Box VMs, necessary for [loopback]
-#   IPv6 support
-
```
Comment (PastaPastaPasta) [inline .travis.yml:4]: 14770 dont do this

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]
14228 needs wrapping https://github.com/dashpay/dash/pull/4313/commits/06d1dfabde3bb42a880dae91388c767008fdb4eb#diff-1d6f5264407b3b2e18be8d9d9653bbf1d67979f4f95dc26575dc23bb8463b4e7R280-R281 like https://github.com/bitcoin/bitcoin/pull/14228/files#diff-1d6f5264407b3b2e18be8d9d9653bbf1d67979f4f95dc26575dc23bb8463b4e7L214-R225

File: src/qt/bitcoingui.cpp [hunk 6]
```diff
@@ -834,11 +836,16 @@ void BitcoinGUI::setWalletActionsEnabled(bool enabled)
 
 void BitcoinGUI::createTrayIcon(const NetworkStyle *networkStyle)
 {
+    assert(QSystemTrayIcon::isSystemTrayAvailable());
+
+#ifndef Q_OS_MAC
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:841]: was not in 14228
```suggestion
```

File: src/qt/bitcoingui.cpp [hunk 7]
```diff
@@ -834,11 +836,16 @@ void BitcoinGUI::setWalletActionsEnabled(bool enabled)
...
     trayIcon = new QSystemTrayIcon(this);
     QString toolTip = tr("%1 client").arg(tr(PACKAGE_NAME)) + " " + networkStyle->getTitleAddText();
     trayIcon->setToolTip(toolTip);
     trayIcon->setIcon(networkStyle->getTrayAndWindowIcon());
     trayIcon->hide();
+#endif
+
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:848]:
```suggestion
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/qt/bitcoingui.cpp [hunk 8]
```diff
@@ -834,11 +836,14 @@ void BitcoinGUI::setWalletActionsEnabled(bool enabled)
...
 void BitcoinGUI::createTrayIcon(const NetworkStyle *networkStyle)
 {
+    assert(QSystemTrayIcon::isSystemTrayAvailable());
+
     trayIcon = new QSystemTrayIcon(this);
     QString toolTip = tr("%1 client").arg(tr(PACKAGE_NAME)) + " " + networkStyle->getTitleAddText();
     trayIcon->setToolTip(toolTip);
     trayIcon->setIcon(networkStyle->getTrayAndWindowIcon());
     trayIcon->hide();
+
```
Comment (UdjinM6) [inline src/qt/bitcoingui.cpp:846]: 14228: nit
```suggestion
```

**Review by UdjinM6 — COMMENTED** [review]
One tiny issue, LGTM otherwise

**Review by UdjinM6 — APPROVED** [review]
LGTM, utACK

File: src/qt/optionsdialog.cpp [hunk 9]
```diff
@@ -302,6 +310,11 @@ void OptionsDialog::setMapper()
     mapper->addMapping(ui->proxyIpTor, OptionsModel::ProxyIPTor);
     mapper->addMapping(ui->proxyPortTor, OptionsModel::ProxyPortTor);
 
+    if (QSystemTrayIcon::isSystemTrayAvailable()) {
+        mapper->addMapping(ui->hideTrayIcon, OptionsModel::HideTrayIcon);
+        mapper->addMapping(ui->minimizeToTray, OptionsModel::MinimizeToTray);
+    }
+
```
Comment (PastaPastaPasta) [inline src/qt/optionsdialog.cpp:317]: 14228 I believe this is a problem. 

Should be done here: https://github.com/dashpay/dash/pull/4313/commits/d5372579def7ebce3281241f03e8582a7ee69bdd#diff-1d6f5264407b3b2e18be8d9d9653bbf1d67979f4f95dc26575dc23bb8463b4e7R280-R281 IE line 280-281

**Review by PastaPastaPasta — CHANGES_REQUESTED** [review]

**Review by PastaPastaPasta — APPROVED** [review]
LGTM, utACK for merge via merge commit

**Pull request merged**